### PR TITLE
Display exclamation point when a flag is overriden

### PR DESCRIPTION
**Pull request by GreatWizard**

### CODE REVIEW

**Comment by GreatWizard** [comment]
Elvina is working on integration tests, I'm waiting her PR about tests :)

**Comment by GreatWizard** [comment]
Updated with integration tests :)

@saintsebastian maybe it will helps you for write others ;) I setup a good features service mock

File: addon/templates/components/feature-controls.hbs [hunk 1]
```diff
@@ -11,7 +11,14 @@
       <tr>
         <td>{{flag.key}}</td>
         <td>{{flag.description}}</td>
-        <td>{{input type='checkbox' checked=flag.isEnabled change=(action 'doToggleFeature' flag.key flag.isEnabled) data-test-checkbox-flag=flag.key}}</td>
+        <td>
+          <label data-test-label-flag={{flag.key}}>
+            {{input type='checkbox' checked=flag.isEnabled change=(action 'doToggleFeature' flag.key flag.isEnabled) data-test-checkbox-flag=flag.key}}
+            {{#if (not-eq flag.isEnabled flag.default)}}
+              ❗
```
Comment (amessinger) [inline addon/templates/components/feature-controls.hbs]: Would you like to add some explanation? `<span title="This flag was overriden">❗</span>`

**Comment by xcambar** [comment]
How about checking the value is updated in the service mock in your integration tests?
As in `assert.ok(this.get('features').isEnabled('flagTrue'))` for instance.

File: tests/integration/components/feature-controls-test.js [hunk 2]
```diff
@@ -1,16 +1,93 @@
 import { module, test } from 'qunit';
 import { setupRenderingTest } from 'ember-qunit';
-import { render } from '@ember/test-helpers';
+import { render, click } from '@ember/test-helpers';
 import hbs from 'htmlbars-inline-precompile';
+import { camelize } from '@ember/string';
+
+const featuresMock = {
+  flags: [camelize('flag-true'), camelize('flag-false')],
+  isEnabled(key) {
```
Comment (saintsebastian) [inline tests/integration/components/feature-controls-test.js]: probably can be just `return key === camelize('flag-true')`

File: addon/templates/components/feature-controls.hbs (hunk 1 above)
Comment (GreatWizard) [reply addon/templates/components/feature-controls.hbs]: :+1:

Comment (saintsebastian) [reply addon/templates/components/feature-controls.hbs]: +1

File: tests/integration/components/feature-controls-test.js [hunk 3]
```diff
@@ -1,16 +1,93 @@
...
+    assert.equal(
+      this.element
+        .querySelector('[data-test-label-flag="flagFalse"]')
+        .textContent.trim(),
+      ''
+    );
+    await click('[data-test-checkbox-flag="flagTrue"]');
+    assert.notOk(
+      this.element.querySelector('[data-test-checkbox-flag="flagTrue"]')
+        .checked
```
Comment (saintsebastian) [inline tests/integration/components/feature-controls-test.js]: you can test that enabled/disabled action has been called here

**Comment by GreatWizard** [comment]
Updated thanks to your feedbacks

**Comment by GreatWizard** [comment]
I adress every issues :) Let's go, thanks everyone

**Pull request merged**